docs/adr/0003-delivery-channels-templating.md

```diff
@@ -0,0 +1,88 @@
+## Decision
+
+A notification having necessary information needs to be translated into a certain communication object (SMS, Email, etc) before it can be sent via the desired delivery channel (SendGrid, Twillio, Airship).
+
+An example of this would be a SMS, which needs the essential information to be available for it to be converted into an SMS object before being sent to the end user.
+
+```c#
+
+    public class SMS {
+        public string ToPhoneNumber { get; set; }
+        public string Body { get; set; }
+    }
+
+```
+
+Properties in the communication object will be populated through different sources based on the communication type.
+
+Eg. Properties in the above SMS object will be populated from the user preference store (azure table storage) from which the `ToPhoneNumber` will be consumed while the `Body` will be populated from the template store (azure table storage).
+
+### Creation of templates, how does someone create templates and in what language
+
+A template will be created based on the notification type, the communication type, and the communication object's property key.
+
+A few examples of this would be:
+
+```
+(Notification type) -> (Communication type) -> (Communication type property)
+
+New Lead            -> SMS                  -> Body
+
+New Lead            -> Email                -> Subject
+New Lead            -> Email                -> Body
+
+```
+
+Every communication type will have a similar communication object, ie, an SMS will use an SMSobject that has certain set of fields. Some of those fields would be required without which a notification would be impossible to be sent. Eg. A SMSObject should have `ToPhoneNumber` or EmailObject should have `ToEmail` and without these properties being populated a notification could not be sent. Furthermore, the `Body` for these 2 communication object would be equally important as there would be no point in sending an empty notification.
+
+Creation of templates can be totally flexible and dependent on the organisation which will integrate with `Marain.UserNotifications` thus every tenant will have their own set of templates for SMS, Email etc. if that organisation wishes to use that communication channel and would not need to setup templates for the unused communication channels.
+
+So, if a notification is created but its corresponding configuration is not setup for that tenant and its communication type, then there will be no notification triggered and an error would be logged in the existing notification data table (azure storage table) which would alert that there is some missing configuration for the newly added notification type for the choosen delivery channel.
+
+### Storage of templates
+
+Templates need to be stored in such a way where it can be accessed and modified easily on the fly. Thus to have this flexibilty, templates will be stored in azure table storage. The template for individual properties (`Body`, `Subject`) would have the following column headers:
```

Comment:
What does "on the fly" mean? Are we expecting templates to be modified as part of sending notifications?